### ISSUE TITLE
`checkCompletionStatusFor` completes when all child models are unavailable

### ISSUE THREAD

**Issue by danielghost** [adaptlearning/adapt-contrib-core#448]
As `Array.every` returns true when invoked on an empty array, the following lines will set models to `_isComplete: true` if all child models are unavailable (`_isAvailable: false`):
https://github.com/adaptlearning/adapt-contrib-core/blob/master/js/models/adaptModel.js#L319
https://github.com/adaptlearning/adapt-contrib-core/blob/master/js/models/adaptModel.js#L326-L328

**Comment by oliverfoster** [comment]
It's not so much an issue with `every` necessarily, perhaps more that we haven't defined the behaviour of completion when there are no `_isAvailable = true` children, which should probably become its own return early statement.

Are we assuming that an "empty" parent should not automatically complete?

**Comment by danielghost** [comment]
Yes, the problem can be seen when using a plugin which modifies the available children, like a role selector. If a user switches roles, the models which previously had all its children set to `_isAvailable: false` will be marked as completed.

Preventing completion on models without available children solves this issue, but it can cause a potential problem with trickle where the trickle button is added to an optional last block, which would have previously not been added as the block would be set to completed. Setting `"_showEndOfPage": false` will resolve this, but requires either a change to the default value, or for the JSON config to be amended accordingly for affected courses.
